java/src/main/java/ai/rapids/cudf/JSONOptions.java

```diff
@@ -18,24 +18,78 @@
 
 package ai.rapids.cudf;
 
+import java.util.Collection;
+
 /**
  * Options for reading in JSON encoded data.
  */
-public final class JSONOptions  extends ColumnFilterOptions {
+public final class JSONOptions extends ColumnFilterOptions {
 
   public static JSONOptions DEFAULT = new JSONOptions(builder());
 
+  private final boolean dayFirst;
+  private final boolean lines;
+
   private JSONOptions(Builder builder) {
     super(builder);
+    dayFirst = builder.dayFirst;
+    lines = builder.lines;
+  }
+
+  public boolean isDayFirst() {
+    return dayFirst;
+  }
+
+  public boolean isLines() {
+    return lines;
+  }
+
+  @Override
+  String[] getIncludeColumnNames() {
+    throw new IllegalArgumentException("JSON reader didn't support column prune");
```

Comment:
here and other places: what is an illegal argument here? should it be UnsupportedOperationException?

Reply:
Done. Thx